src/classes/CaseTriggerHandler.cls

```diff
@@ -0,0 +1,135 @@
+/**
+ * Created by Synebo on 14.09.2021.
+ */
+
+public with sharing class CaseTriggerHandler extends TriggerHandler {
+
+    public override void afterUpdate() {
+        this.countCaseUpdate((List<Case>) Trigger.new, (Map<Id, Case>) Trigger.oldMap);
+    }
+
+    public override void afterInsert() {
+        this.countCaseInsert((List<Case>) Trigger.new);
+    }
+
+    public override void afterUndelete() {
+        this.countCaseInsert((List<Case>) Trigger.new);
+    }
+    public override void afterDelete() {
+        this.countCaseDelete((Map<Id, Case>) Trigger.oldMap);
+    }
+
+    private void countCaseDelete(Map<Id, Case>oldMap){
```

Comment:
countCaseDelete, countCaseInsert and countCaseUpdate can be as a single method with right condition